lib/std/os/linux.zig

```diff
@@ -9089,6 +9107,125 @@ pub const perf_event_attr = extern struct {
     sample_max_stack: u16 = 0,
     /// Align to u64
     __reserved_2: u16 = 0,
+
+    aux_sample_size: u32 = 0,
+    aux_action: packed struct(u32) {
+        /// start AUX area tracing paused
+        start_paused: bool = false,
+        /// on overflow, pause AUX area tracing
+        pause: bool = false,
+        /// on overflow, resume AUX area tracing
+        @"resume": bool = false,
+
+        __reserved_3: u29 = 0,
+    } = .{},
+
+    /// User provided data if sigtrap == true
+    sig_data: u64 = 0,
+
+    /// Extension of config2
+    config3: u64 = 0,
+};
+
+pub const perf_event_header = extern struct {
+    /// Event type: sample/mmap/fork/etc.
+    type: PERF.RECORD,
+    /// Additional informations on the event: kernel/user/hypervisor/etc.
+    misc: packed struct(u16) {
+        cpu_mode: enum(u3) {
+            UNKNOWN = 0,
+            KERNEL = 1,
+            USER = 2,
+            HYPERVISOR = 3,
+            GUEST_KERNEL = 4,
+            GUEST_USER = 5,
+        },
+        _: u9,
+        PROC_MAP_PARSE_TIMEOUT: bool,
+        bit13: packed union {
```

Comment:
```suggestion
        bit13: packed union(u1) {
```

#24715

Reply:
Just to check: `packed union(u1)` doesn’t compile on current Zig master. The linked issue looks like just a proposal, while the syntax itself comes from #19754, which is accepted but not implemented yet (and I don’t see a PR for it). Wouldn’t that leave this patch unbuildable until #19754 actually lands? Since these unions sit in a packed struct anyway, if they are not consume a bit the struct size should still fail to match and trigger a compile error?

Reply:
Oh, yes, you're right. Just ignore these two suggestions.